### ISSUE TITLE
CI jobs are randomly crashing

### ISSUE THREAD

**Issue by smutch** [mccoy-devs/mccoy#52]
Not sure why this is. It seems to happen randomly but re-running manually often allows the tests to pass...

**Comment by smutch** [comment]
Seems to have something to do with the beagle extensions...

``` txt
#
[568](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:569)
# A fatal error has been detected by the Java Runtime Environment:
[569](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:570)
#
[570](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:571)
#  SIGILL (0x4) at pc=0x00007f42509bcc9b, pid=2787, tid=0x00007f427dcc6700
[571](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:572)
#
[572](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:573)
# JRE version: OpenJDK Runtime Environment (8.0_144-b01) (build 1.8.0_144-b01)
[573](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:574)
# Java VM: OpenJDK 64-Bit Server VM (25.144-b01 mixed mode linux-amd64 compressed oops)
[574](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:575)
# Problematic frame:
[575](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:576)
# C  [libhmsbeagle.so+0x5c9b]  beagleLoadPlugins()+0x1db
[576](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:577)
#
[577](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:578)
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
[578](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:579)
#
[579](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:580)
# An error report file with more information is saved as:
[580](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:581)
# /tmp/pytest-of-runner/pytest-0/test_beast0/expected/runs/run_1/hs_err_pid2787.log
[581](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:582)
#
[582](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:583)
# If you would like to submit a bug report, please visit:
[583](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:584)
#   http://www.azulsystems.com/support/
[584](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:585)
# The crash happened outside the Java Virtual Machine in native code.
[585](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:586)
# See problematic frame for where to report the bug.
[586](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:587)
#
[587](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:588)
Full Traceback (most recent call last):
[588](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:589)
  File "/home/runner/work/mccoy/mccoy/.nox/test-python/lib/python3.7/site-packages/snakemake/executors/__init__.py", line 2570, in run_wrapper
[589](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:590)
    runtime_sourcecache_path,
[590](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:591)
  File "/home/runner/work/mccoy/mccoy/.nox/test-python/lib/python3.7/site-packages/mccoy/workflow/rules/beast.smk", line 138, in __rule_beast
[591](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:592)
  File "/home/runner/work/mccoy/mccoy/.nox/test-python/lib/python3.7/site-packages/snakemake/shell.py", line 296, in __new__
[592](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:593)
    raise sp.CalledProcessError(retcode, cmd)
[593](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:594)
subprocess.CalledProcessError: Command 'source /usr/share/miniconda3/bin/activate '/home/runner/work/mccoy/mccoy/tests/.conda/fc39a85f7bb1e12558859c670056f80f'; module purge && module load gcccore/8.3.0 fosscuda/2019b beast/2.6.6; set -euo pipefail;  
[594](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:595)
        if [[ -n "" ]]; then cp  results/beast/expected-1-beast.xml.state; fi
[595](https://github.com/mccoy-devs/mccoy/runs/8008739792?check_suite_focus=true#step:9:596)
        beast -D 'alignment=results/aligned/expected-1.fasta,tracelog=results/beast/expected-1-trace.log,treelog=results/beast/expected-1-tree.log,mcmc.threads=1,ePopSize=0.3,Date.dateFormat=yyyy/MM/dd,Date.delimiter=_,mutationRate=1.0,mcmc.chainLength=10000,treelog.logEvery=1000,tracelog.logEvery=1000' -beagle_GPU -statefile results/beast/expected-1-beast.xml.state results/beast/expected-1-dynamic_beast.xml 1>&2 2> logs/expected-1_beast.log' returned non-zero exit status 134.
```

**Comment by smutch** [comment]
Closed by 2f887546f6f0c4ef6cbaa4984f3c61b455393f33 🤞🏻

**Comment by smutch** [comment]
So this is still not working!

Based on team discussions I will try the following:

- [ ] Set `sampleFromPrior`
- [ ] Set the seed for the run and see if I can find one that works

**Comment by smutch** [comment]
Closed by #54

**Comment by CoolEvilgenius** [comment]
So, have you settle this problem? I am facing the same problem......Thanks.

**Comment by smutch** [comment]
As far as I know this has been solved by #54. The issue was that the most recent version of Beagle available to Conda was incompatible with the default Github CI runners. Downgrading to `beagle-lib<4` seemed to do the trick.

**Comment by smutch** [comment]
Can you point me to the logs from a failed McCoy CI run if this has popped up again as an issue?
